lib/extensions/ar_migration.rb

```diff
@@ -0,0 +1,97 @@
+module ArPglogicalMigration
+  module PglogicalMigrationHelper
+    def self.migrations_column_present?
+      ActiveRecord::Base.connection.columns("miq_regions").any? { |c| c.name == "migrations_ran" }
+    end
+
+    def self.my_region_number
+      @my_region_number ||= ApplicationRecord.my_region_number
```

Comment:
I overheard you describing how you can use ApplicationRecord here but must must use AR::Base elsewhere because reasons... the flip flopping of their usage here seems to need a "why" comment

Reply:
So the only places I *need* to use `ActiveRecord::Base` are when the code is evaluated at load-time rather than runtime. So technically I could use `ApplicationRecord.connection` in the definitions of `.migrations_column_present?` and `.my_region_created?`.

Here I need to use `ApplicationRecord` because we include the region-aware modules into that class rather than `ActiveRecord::Base`.

The place that would cause a problem with constant loading would really be [HelperARClass](https://github.com/ManageIQ/manageiq/pull/17919/files#diff-83583f4cb3747e75378008fb3501293aR36). But I also think I could probably get away with using an anonymous class there if I tried really hard. What do you think?